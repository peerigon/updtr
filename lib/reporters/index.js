"use strict";

module.exports = {
    default: require("./default.js"),
    shy: require("./shy.js"),
    simple: require("./simple.js"),
};
