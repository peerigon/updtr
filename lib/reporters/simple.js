"use strict";

var unicons = require("unicons");

var util = require("./util");

function defaultReporter(emitter) {
    var currentLine = "";

    function logProgress(message) {
        currentLine = message;
        console.log(message + "... ".grey);
    }

    function finishProgress(message) {
        if (message) {
            console.log.apply(console, arguments);
        } else {
            console.log(currentLine);
        }
        currentLine = "";
    }

    function logUpdateProgress(event, status, message) {
        var progress = event.current + "/" + event.total;
        var circle = unicons.cli("circle");
        var name = event.info.name;

        if (status === "error") {
            circle = circle.red;
            name = name.red;
        } else if (status === "success") {
            circle = circle.green;
        } else {
            circle = circle.grey;
        }

        logProgress([
            progress.grey,
            "\t",
            circle,
            name,
            message,
        ].join(" "));
    }

    // Activate colors by extending String.prototype
    require("colors"); // eslint-disable-line import/no-unassigned-import

    emitter.on("init", function () {
        logProgress("Looking for outdated modules");
    });
    emitter.on("noop", function () {
        finishProgress("No outdated modules found. Are they installed?");
    });
    emitter.on("modulesMissing", function (event) {
        finishProgress(util.modulesMissingMessage(event));
    });
    emitter.on("outdated", function (event) {
        finishProgress("Found %s outdated module%s", event.total, event.total === 1 ? "" : "s");
        console.log("");
        console.log("Starting to update your modules" + "...".grey);
    });
    emitter.on("updating", function (event) {
        var info = event.info;

        logUpdateProgress(event, "pending", "updating ".grey + info.current + " " + unicons.cli("arrowRight").grey + " " + info.updateTo);
    });
    emitter.on("testing", function (event) {
        logUpdateProgress(event, "pending", "testing".grey);
    });
    emitter.on("rollback", function (event) {
        logUpdateProgress(event, "error", "rolling back".grey);
    });
    emitter.on("rollbackDone", function (event) {
        var info = event.info;

        logUpdateProgress(event, "error", info.updateTo + " failed".grey);
        console.log("");
    });
    emitter.on("testStdout", function (event) {
        console.log(event.testStdout);
    });
    emitter.on("updatingDone", function (event) {
        var info = event.info;

        logUpdateProgress(event, "success", info.updateTo + " success".grey);
        console.log("");
    });
    emitter.on("finished", function () {
        console.log("");
        console.log("Finished");
    });
}

module.exports = defaultReporter;
