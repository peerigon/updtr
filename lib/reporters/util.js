"use strict";

function modulesMissingMessage(event) {
    var moduleList = event.infos.map(function (info) {
        return info.name;
    }).join(", ");

    return "Some modules are not installed: " + moduleList +
           ". Please install with 'npm install' first.";
}

module.exports = {
    modulesMissingMessage: modulesMissingMessage,
};

