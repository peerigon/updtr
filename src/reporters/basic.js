import ansiEscapes from "ansi-escapes";
import chalk from "chalk";
import unicons from "unicons";
import Message from "./util/Message";
import Indicator, {
    INDICATOR_NEUTRAL,
    INDICATOR_PENDING,
    INDICATOR_OK,
    INDICATOR_FAIL,
} from "./util/Indicator";
import customConfigToLines from "./util/customConfigToLines";
import pluralize from "./util/pluralize";
import handleError from "./util/handleError";
import msToString from "./util/msToString";
import {
    filterSuccessfulUpdates,
    filterFailedUpdates,
} from "../tasks/util/filterUpdateResults";

function updatingLine(updateTask) {
    return [
        new Indicator(INDICATOR_PENDING),
        chalk.bold(updateTask.name),
        chalk.grey("updating"),
        updateTask.rollbackTo,
        chalk.grey(unicons.arrowRight),
        updateTask.updateTo + chalk.grey("..."),
    ].join(" ");
}

function testingLine(updateTask) {
    return [
        new Indicator(INDICATOR_PENDING),
        chalk.bold(updateTask.name),
        chalk.grey("testing..."),
    ].join(" ");
}

function rollbackLine(updateTask) {
    return [
        new Indicator(INDICATOR_FAIL),
        chalk.bold.red(updateTask.name),
        chalk.grey("rolling back"),
        updateTask.updateTo,
        chalk.grey(unicons.arrowRight),
        updateTask.rollbackTo + chalk.grey("..."),
    ].join(" ");
}

function successLine(updateTask) {
    return [
        new Indicator(INDICATOR_OK),
        chalk.bold(updateTask.name),
        updateTask.updateTo,
        chalk.grey("success"),
    ].join(" ");
}

function failLine(updateTask) {
    return [
        new Indicator(INDICATOR_FAIL),
        chalk.bold.red(updateTask.name),
        updateTask.updateTo,
        chalk.grey("failed"),
    ].join(" ");
}

function excludedLine(excluded) {
    return [
        new Indicator(INDICATOR_NEUTRAL),
        chalk.bold(excluded.name),
        chalk.grey(excluded.reason),
    ].join(" ");
}

function cmdToLines(description, cmd) {
    const lines = Array.isArray(description) === true ?
        description :
        [description];

    return lines.concat([chalk.grey(`> ${ cmd } `)]);
}

function writeLinesToConsole(lines) {
    if (lines.length === 0) {
        return;
    }
    console.log(ansiEscapes.eraseDown + lines.join("\n"));
}

export default function (updtr, reporterConfig) {
    const startTime = Date.now();
    let excludedModules;

    updtr.on("start", ({ config }) => {
        writeLinesToConsole(customConfigToLines(config));
    });
    updtr.on("init/install-missing", ({ cmd }) => {
        writeLinesToConsole(
            cmdToLines(
                "Installing missing dependencies" + chalk.grey("..."),
                cmd
            )
        );
    });
    updtr.on("init/collect", ({ cmd }) => {
        writeLinesToConsole(
            cmdToLines("Looking for outdated modules" + chalk.grey("..."), cmd)
        );
    });
    updtr.on("init/end", ({ updateTasks, excluded }) => {
        excludedModules = excluded;
        if (updateTasks.length === 0 && excluded.length === 0) {
            writeLinesToConsole(["Everything " + chalk.bold("up-to-date")]);
        } else if (updateTasks.length === 0) {
            writeLinesToConsole([
                chalk.bold("No updates available") +
                " for the given modules and version range",
            ]);
        } else {
            writeLinesToConsole([
                new Message("Found " + chalk.bold("%s update%s") + ".", [
                    updateTasks.length,
                    pluralize(updateTasks.length),
                ]),
                "",
            ]);
        }
    });
    updtr.on("batch-update/updating", event => {
        writeLinesToConsole(
            cmdToLines(event.updateTasks.map(updatingLine), event.cmd)
        );
    });
    updtr.on("batch-update/testing", event => {
        writeLinesToConsole(
            cmdToLines(event.updateTasks.map(testingLine), event.cmd)
        );
    });
    updtr.on("batch-update/rollback", event => {
        writeLinesToConsole(
            cmdToLines(event.updateTasks.map(rollbackLine), event.cmd)
        );
    });
    updtr.on("batch-update/result", event => {
        if (event.success === true) {
            writeLinesToConsole(
                event.updateTasks.map(event.success ? successLine : failLine)
            );
        }
        // Not showing the test stdout here when there was an error because
        // we will proceed with the sequential update.
    });
    updtr.on("sequential-update/updating", event => {
        writeLinesToConsole(cmdToLines(updatingLine(event), event.cmd));
    });
    updtr.on("sequential-update/testing", event => {
        writeLinesToConsole(cmdToLines(testingLine(event), event.cmd));
    });
    updtr.on("sequential-update/rollback", event => {
        writeLinesToConsole(cmdToLines(rollbackLine(event), event.cmd));
    });
    updtr.on("sequential-update/result", event => {
        writeLinesToConsole([(event.success ? successLine : failLine)(event)]);
        if (reporterConfig.testStdout === true && event.success === false) {
            writeLinesToConsole([event.stdout]);
        }
    });
    updtr.on("end", ({ results }) => {
        const duration = msToString(Date.now() - startTime);
        const successful = filterSuccessfulUpdates(results);
        const failed = filterFailedUpdates(results);

        writeLinesToConsole([""]);

        if (successful.length > 0) {
            writeLinesToConsole([
                new Message(chalk.bold("%s successful") + " update%s.", [
                    successful.length,
                    pluralize(successful.length),
                ]),
            ]);
        }
        if (failed.length > 0) {
            writeLinesToConsole([
                new Message(chalk.bold("%s failed") + " update%s.", [
                    failed.length,
                    pluralize(failed.length),
                ]),
            ]);
        }
        if (excludedModules.length > 0) {
            const list = excludedModules.map(excludedLine);

            if (successful.length > 0 || failed.length > 0) {
                writeLinesToConsole([""]);
            }
            writeLinesToConsole(
                [
                    new Message(chalk.bold("%s skipped") + " module%s:", [
                        excludedModules.length,
                        pluralize(excludedModules.length),
                    ]),
                    "",
                ].concat(list)
            );
        }

        writeLinesToConsole(["", new Message("Finished after %s.", [duration])]);
    });
    updtr.on("error", err => void handleError(err));
}
