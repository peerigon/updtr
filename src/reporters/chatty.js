import { Spinner } from "cli-spinner";
import unicons from "unicons";
import chalk from "chalk";

export default function chatty(updtr) {
    let currentLine = "";
    let spinner;

    function logProgress(message) {
        if (spinner) {
            spinner.stop();
        }
        currentLine = message;
        spinner = new Spinner(message + chalk.grey.dim("... %s"));
        spinner.setSpinnerString(19);
        spinner.start();
    }

    function finishProgress(message, ...args) {
        if (spinner) {
            spinner.stop(true);
        }
        if (message) {
            console.log(message, ...args);
        } else {
            console.log(currentLine);
        }
        currentLine = "";
    }

    function logUpdateProgress(event, status, message) {
        const progress = event.current + "/" + event.total;
        let circle = unicons.cli("circle");
        let name = event.info.name;

        if (status === "error") {
            circle = circle.red;
            name = name.red;
        } else if (status === "success") {
            circle = circle.green;
        } else {
            circle = circle.grey;
        }

        logProgress([progress.grey, "\t", circle, name, message].join(" "));
    }

    updtr.on("init/start", event => {
        logProgress("Looking for outdated modules");
    });
    updtr.on("end", event => {
        console.log(chalk.bold.green("Finished"));
    });
}
