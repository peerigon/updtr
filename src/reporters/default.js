"use strict";

const Spinner = require("cli-spinner").Spinner;
const unicons = require("unicons");

const util = require("./util");

function defaultReporter(emitter) {
    let currentLine = "";
    let spinner;

    function logProgress(message) {
        if (spinner) {
            spinner.stop();
        }
        currentLine = message;
        spinner = new Spinner(message + "... ".grey + "%s ");
        spinner.setSpinnerString(19);
        spinner.start();
    }

    function finishProgress(message, ...args) {
        if (spinner) {
            spinner.stop(true);
        }
        if (message) {
            console.log(message, ...args);
        } else {
            console.log(currentLine);
        }
        currentLine = "";
    }

    function logUpdateProgress(event, status, message) {
        const progress = event.current + "/" + event.total;
        let circle = unicons.cli("circle");
        let name = event.info.name;

        if (status === "error") {
            circle = circle.red;
            name = name.red;
        } else if (status === "success") {
            circle = circle.green;
        } else {
            circle = circle.grey;
        }

        logProgress([
            progress.grey,
            "\t",
            circle,
            name,
            message,
        ].join(" "));
    }

    // Activate colors by extending String.prototype
    require("colors"); // eslint-disable-line import/no-unassigned-import

    emitter.on("init", () => {
        logProgress("Looking for outdated modules");
    });
    emitter.on("noop", () => {
        finishProgress("No outdated modules found. Are they installed?");
    });
    emitter.on("modulesMissing", (event) => {
        finishProgress(util.modulesMissingMessage(event));
    });
    emitter.on("outdated", (event) => {
        finishProgress("Found %s outdated module%s", event.total, event.total === 1 ? "" : "s");
        console.log("");
        console.log("Starting to update your modules" + "...".grey);
    });
    emitter.on("updating", (event) => {
        const info = event.info;

        logUpdateProgress(event, "pending", "updating ".grey + info.current + " " + unicons.cli("arrowRight").grey + " " + info.updateTo);
    });
    emitter.on("testing", (event) => {
        logUpdateProgress(event, "pending", "testing".grey);
    });
    emitter.on("rollback", (event) => {
        logUpdateProgress(event, "error", "rolling back".grey);
    });
    emitter.on("rollbackDone", (event) => {
        const info = event.info;

        logUpdateProgress(event, "error", info.updateTo + " failed".grey);
        finishProgress();
    });
    emitter.on("testStdout", (event) => {
        console.log(event.testStdout);
    });
    emitter.on("updatingDone", (event) => {
        const info = event.info;

        logUpdateProgress(event, "success", info.updateTo + " success".grey);
        finishProgress();
    });
    emitter.on("finished", () => {
        console.log("");
        console.log("Finished");
    });
}

module.exports = defaultReporter;
