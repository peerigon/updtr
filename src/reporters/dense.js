import chalk from "chalk";
import unicons from "unicons";
import Projector from "./util/Projector";
import Terminal from "./util/Terminal";
import Message from "./util/Message";
import Spinner from "./util/Spinner";
import Indicator, {
    INDICATOR_NEUTRAL,
    INDICATOR_PENDING,
    INDICATOR_OK,
    INDICATOR_FAIL,
} from "./util/Indicator";
import customConfigToLines from "./util/customConfigToLines";
import pluralize from "./util/pluralize";
import handleError from "./util/handleError";
import msToString from "./util/msToString";
import {
    filterSuccessfulUpdates,
    filterFailedUpdates,
} from "../tasks/util/filterUpdateResults";

const spinner = new Spinner("dots");

function updatingLine(updateTask) {
    return [
        new Indicator(INDICATOR_PENDING),
        chalk.bold(updateTask.name),
        chalk.grey("updating"),
        updateTask.rollbackTo,
        chalk.grey(unicons.arrowRight),
        updateTask.updateTo + chalk.grey("..."),
    ].join(" ");
}

function testingLine(updateTask) {
    return [
        new Indicator(INDICATOR_PENDING),
        chalk.bold(updateTask.name),
        chalk.grey("testing..."),
    ].join(" ");
}

function rollbackLine(updateTask) {
    return [
        new Indicator(INDICATOR_FAIL),
        chalk.bold.red(updateTask.name),
        chalk.grey("rolling back"),
        updateTask.updateTo,
        chalk.grey(unicons.arrowRight),
        updateTask.rollbackTo + chalk.grey("..."),
    ].join(" ");
}

function successLine(updateTask) {
    return [
        new Indicator(INDICATOR_OK),
        chalk.bold(updateTask.name),
        updateTask.updateTo,
        chalk.grey("success"),
    ].join(" ");
}

function failLine(updateTask) {
    return [
        new Indicator(INDICATOR_FAIL),
        chalk.bold.red(updateTask.name),
        updateTask.updateTo,
        chalk.grey("failed"),
    ].join(" ");
}

function excludedLine(excluded) {
    return [
        new Indicator(INDICATOR_NEUTRAL),
        chalk.bold(excluded.name),
        chalk.grey(excluded.reason),
    ].join(" ");
}

function cmdToLines(description, cmd) {
    const lines = Array.isArray(description) === true ?
        description :
        [description];

    return lines.concat([chalk.grey(`> ${ cmd } `), spinner]);
}

export default function (updtr, reporterConfig) {
    const terminal = new Terminal(reporterConfig.stream);
    const projector = new Projector(terminal);
    const startTime = Date.now();
    let excludedModules;

    updtr.on("start", ({ config }) => {
        terminal.append(customConfigToLines(config));
    });
    updtr.on("init/install-missing", ({ cmd }) => {
        projector.display(
            cmdToLines(
                "Installing missing dependencies" + chalk.grey("..."),
                cmd
            )
        );
    });
    updtr.on("init/collect", ({ cmd }) => {
        projector.display(
            cmdToLines("Looking for outdated modules" + chalk.grey("..."), cmd)
        );
    });
    updtr.on("init/end", ({ updateTasks, excluded }) => {
        excludedModules = excluded;
        projector.stop();
        if (updateTasks.length === 0 && excluded.length === 0) {
            terminal.append(["Everything " + chalk.bold("up-to-date")]);
        } else if (updateTasks.length === 0) {
            terminal.append([
                chalk.bold("No updates available") +
                    " for the given modules and version range",
            ]);
        } else {
            terminal.append([
                new Message("Found " + chalk.bold("%s update%s") + ".", [
                    updateTasks.length,
                    pluralize(updateTasks.length),
                ]),
                "",
            ]);
        }
    });
    updtr.on("batch-update/updating", event => {
        projector.display(
            cmdToLines(event.updateTasks.map(updatingLine), event.cmd)
        );
    });
    updtr.on("batch-update/testing", event => {
        projector.display(
            cmdToLines(event.updateTasks.map(testingLine), event.cmd)
        );
    });
    updtr.on("batch-update/rollback", event => {
        projector.display(
            cmdToLines(event.updateTasks.map(rollbackLine), event.cmd)
        );
    });
    updtr.on("batch-update/result", event => {
        projector.stop();
        if (event.success === true) {
            terminal.append(
                event.updateTasks.map(event.success ? successLine : failLine)
            );
        }
        // Not showing the test stdout here when there was an error because
        // we will proceed with the sequential update.
    });
    updtr.on("sequential-update/updating", event => {
        projector.display(cmdToLines(updatingLine(event), event.cmd));
    });
    updtr.on("sequential-update/testing", event => {
        projector.display(cmdToLines(testingLine(event), event.cmd));
    });
    updtr.on("sequential-update/rollback", event => {
        projector.display(cmdToLines(rollbackLine(event), event.cmd));
    });
    updtr.on("sequential-update/result", event => {
        projector.stop();
        terminal.append([(event.success ? successLine : failLine)(event)]);
        if (reporterConfig.testStdout === true && event.success === false) {
            terminal.append([event.stdout]);
        }
    });
    updtr.on("end", ({ results }) => {
        const duration = msToString(Date.now() - startTime);
        const successful = filterSuccessfulUpdates(results);
        const failed = filterFailedUpdates(results);

        terminal.append([""]);

        if (successful.length > 0) {
            terminal.append([
                new Message(chalk.bold("%s successful") + " update%s.", [
                    successful.length,
                    pluralize(successful.length),
                ]),
            ]);
        }
        if (failed.length > 0) {
            terminal.append([
                new Message(chalk.bold("%s failed") + " update%s.", [
                    failed.length,
                    pluralize(failed.length),
                ]),
            ]);
        }
        if (excludedModules.length > 0) {
            const list = excludedModules.map(excludedLine);

            if (successful.length > 0 || failed.length > 0) {
                terminal.append([""]);
            }
            terminal.append(
                [
                    new Message(chalk.bold("%s skipped") + " module%s:", [
                        excludedModules.length,
                        pluralize(excludedModules.length),
                    ]),
                    "",
                ].concat(list)
            );
        }

        terminal.append(["", new Message("Finished after %s.", [duration])]);
    });
    updtr.on("error", err => void handleError(err));
}
