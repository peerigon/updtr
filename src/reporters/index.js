import dense from "./dense";
import none from "./none";

// The first property here is the default reporter
export default { dense, none };
