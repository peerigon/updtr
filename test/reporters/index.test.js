import reporters from "../../src/reporters";

describe("reporters", () => {
    it("should export all available reporters", () => {
        expect(Object.keys(reporters)).toMatchSnapshot();
    });
});
