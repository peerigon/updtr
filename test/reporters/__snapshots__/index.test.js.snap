// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reporters should export all available reporters 1`] = `
Array [
  "dense",
  "none",
]
`;
